### ISSUE TITLE
node.js with sails.js pm2 socket.io reconnect

### ISSUE THREAD

**Issue by alex986** [balderdashy/sails#2916]
I use node v0.12.2, sails.js v0.11 and pm2 0.12.13, in the browser appear 'Connection closed before receiving a handshake response' and 'Error: xhr post error'. How can I fix this?

config.session

adapter: 'redis',
host: '127.0.0.1',
port: 6379,
ttl: 0,
db: 'sails'
config.socket

adapter: 'socket.io-redis',
host: '127.0.0.1',
port: 6379,
db: 'sails'

**Comment by tjwebb** [comment]
I'm guessing you're using clustering with pm2? Please post your full sockets and session config files (in a gist, or link to them if project is open-source)

**Comment by alex986** [comment]
These are the files. https://gist.github.com/alex986/7a3acaf29c26da2645c5

**Comment by alex986** [comment]
In windows work fine but in Linux dont work. Centos 6.6, kernel 2.6.32-504.16.2.el6.x86_64.
![sin titulo](https://cloud.githubusercontent.com/assets/5686138/7578638/2271d496-f81d-11e4-988c-ed44bdfb5471.png)

**Comment by dapriett** [comment]
I'm seeing the same thing, per my issue here: https://github.com/balderdashy/sails/issues/2927

Does anything or pm2 (or another clustering module) working with sails?  I'm using Redis for sessions and sockets.

**Comment by sailsbot** [comment]
Thanks for posting, @alex986.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
